gateway/src/main/kotlin/com/example/gateway/client/NatsClient.kt

```diff
@@ -17,4 +24,22 @@ class NatsClient(private val natsConnection: Connection) {
         return Mono.fromFuture { natsConnection.request(subject, payload.toByteArray()) }
             .map { response -> parser.parseFrom(response.data) }
     }
+
+    fun subscribeByUserId(userId: String): Flux<UpdateDeviceResponse> {
```

Comment:
strange signature: method name defines that you use pub-sub logic, but the return type relates to req-rep logic